src/CssParsing/StyleRule.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\CssParsing;
+
+/**
+ * This class represents a CSS style rule, including a media query, selectors and a declarations block.
+ *
+ * @internal
+ */
+class StyleRule
+{
+    /**
+     * @var string
+     */
+    private $mediaQuery;
```

Comment:
Technically it's not just the media query, it also contains the `"@media "` preceding it.

But I think we can be more generic here (see #1078).  It could be any enclosing at-rule.  Based on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/At-rule)'s nomenclature, it's 'at, identifier and rule'.  In context, it's the 'enclosing', 'parent' or 'nesting' one of those.

Reply:
I went with `containingAtRule` in #1015, though am not completely happy with that.  It implies it's an object when in fact it's just a string.  But I'm not doing Microsoft-style incorporation of types into variable names, whether pre- or post-fix.  There should be a way to describe that part of the rule (without it's nested content).

Reply:
Done.